### PR TITLE
add fake_from_yaml method to StorageManger

### PR DESCRIPTION
**Pull request by wfeldt**

### CODE REVIEW

**Comment by ancorgs** [comment]
Currently this adds some code duplication with `FakeProbing`, so I would like to see this code coming in and `FakeProbing` dying as part of the pull request.

**Comment by shundhammer** [comment]
Looks good so far. But the examples and clients seem to be missing.

**Comment by shundhammer** [comment]
LGTM now.

**Comment by ancorgs** [comment]
Travis and jenkins don't look happy. Even coveralls is complaining (but not sure if coveralls claims are always legitimate).

File: test/boot_requirements_checker_test.rb [hunk 1]
```diff
@@ -40,7 +40,7 @@ def find_vol(mount_point, volumes)
     let(:storage_arch) { instance_double("::Storage::Arch") }
 
     before do
-      Yast::Storage::FakeProbing.new
+      Yast::Storage::StorageManager.fake_from_yaml
```
Comment (ancorgs) [inline test/boot_requirements_checker_test.rb]: Hmm, calling `fake_from_yaml` without yaml is a quite hacky and unexpected usage. `fake_from_yaml` actually does two things - creating a fake instance of the StorageManager and populating `probed` with yaml.

For cases like this, in which we only want to do the first part but not the second one, we should have another method.

Like already suggested here, for example, https://github.com/yast/yast-storage/pull/193#discussion_r53643452

File: src/lib/storage/storage_manager.rb [hunk 2]
```diff
@@ -58,6 +59,26 @@ def instance
...
         end
 
+        # Use this as an alternative to the instance method.
+        # Probing is skipped and the device tree is initialized from yaml_file.
+        # Any existing probed device tree is replaced.
+        #
+        # @return [::Storage::Storage] libstorage object
+        #
+        def fake_from_yaml(yaml_file = nil)
+          create_instance(
```
Comment (ancorgs) [inline src/lib/storage/storage_manager.rb]: This call to `create_instance` should be in a separated public method IMHO. See below for cases in which is needed.

File: test/boot_requirements_checker_test.rb (hunk 1 above)
Comment (wfeldt) [inline test/boot_requirements_checker_test.rb]: calling fake_from_yaml(nil) seems an obvious semantic to me, but rubocop won't let me

my first thought was to call the method just 'fake'; maybe that is more understandable after all

Comment (wfeldt) [inline test/boot_requirements_checker_test.rb]: > For cases like this, in which we only want to do the first part but not the second one,
> we should have another method

I can't follow you here. That's like arguing against constructors with arguments.

fake() resp. fake_from_yaml() creates a fake instance (and populates it with the arg)

File: src/lib/storage/storage_manager.rb (hunk 2 above)
Comment (wfeldt) [inline src/lib/storage/storage_manager.rb]: I think not. The current semantics works quite well; this patch shows that the code gets much simpler.

**Comment by wfeldt** [comment]
merging, you can shoot me later...

Actually, I think renaming fake_from_yaml -> fake might be a good idea and address part of your concerns.

**Pull request merged**